assets/userCtrl.js

```diff
@@ -274,6 +301,12 @@ module.exports = function ($scope, $location, io) {
         if (illegal_match) {
           return {correct: false, error: 'Your in-game name contains an illegal character: ' + illegal_match};
         }
+          
+        // Since Reddit's flair_text system is emoji based, we want to prevent users from submitting an IGN with colons to prevent conflicts.
+        var has_colon = game.ign.match(/:/);
+        if (has_colon) {
+          return {correct: false, error: 'Your in-game name contains a colon. Please omit the colons to prevent conflicts with emoji-based flair text.'};
+        }
```

Comment:
I think it would be better to do this in a slightly different way:

* On the client, generate a flair text that doesn't include the emoji, and send that to the server.
* The server (in the `setText` method of `FlairController.js` validates that the flair text doesn't contain any colons.
* The server checks the user's current flair from their css class, adds the appropriate emoji to the flair text, and changes the flair.

The problem with the current way of doing it is that the server is relying on the client to do validation, which isn't good from a security perspective. A user could use (e.g.) devtools in a browser to skip the client-side code and send an unvalidated flair text directly to the server. For example, they could give themselves any flair emoji by just putting that emoji in the flair text.

Reply:
Okay! I'll add a server-side check, just wasn't sure where to put it. I wanted to have a client side validation as well because colons are allowed in-game so I figured it would be more reasonable to give them a message as to why colons aren't allowed.

Reply:
Done!